documentation/src/main/asciidoc/manual-index.asciidoc

```diff
@@ -398,6 +401,114 @@ long executionId = BatchRuntime
 ----
 ====
 
+==== Job Parameters
+
+.Job Parameters in JSR 352 Integration
+|===
+|Parameter Name |Builder Method |Requirement |Default value |Description
+
+|`rootEntities`
+|`forEntity(Class<?>)`, `forEntities(Class<?>, Class<?>...)`
+|Required
+|-
+|The root entity types to index in this job execution.
+
+|`purgeAllOnStart`
+|`purgeAllOnStart(boolean)`
+|Optional
+|False
+|Specify whether the existing lucene index should be purged at the beginning of the job. This
+operation takes place before the step of lucene document production.
+
+|`optimizeAfterPurge`
+|`optimizeAfterPurge(boolean)`
+|Optional
+|False
+|Specify whether the mass indexer should be optimized at the beginning of the job. This operation
+takes place after the purge operation and before the step of lucene document production.
+
+|`optimizeOnFinish`
+|`optimizeOnFinish(boolean)`
+|Optional
+|False
+|Specify whether the mass indexer should be optimized at the end of the job. This operation takes
+place after the step of lucene document production.
+
+|`cacheable`
+|`cacheable(boolean)`
+|Optional
+|False
+|Specify whether the Hibernate queries are cacheable. This setting will be applied to the internal
+entity reader class. Set it to true when reading a complex graph with relations.
+
+|`fetchSize`
+|`fetchSize(int)`
+|Optional
+|200,000
+|The size for the result fetching.
+
+|`customQueryHQL`
+|`restrictedBy(String)`
+|Optional
+|-
+|Use HQL / JPQL to select the entities to index. Mixing this approach with the criteria restriction
+is not allowed. Please notice that there's no query validation for your input.
+
+|`customQueryCriteria`
+|`restrictedBy(Criterion)`
+|Optional
+|-
+|Add criterion to construct a customized selection of mass-indexing under the criteria approach. You
+can call this method as many times as the number of criterion wished. However, mixing this approach
+with the HQL restriction is not allowed.
+
+|`customQueryLimit`
+|`customQueryLimit(int)`
+|Optional
+|1,000,000
+|The limit of number of entities read in a custom query. This parameter is only used under
+custom query approaches (HQL and criteria). The value defined must be greater than 0.
+
+|`checkpointInterval`
+|`checkpointInterval(int)`
+|Optional
+|200
+|The number of entities read before trigger a checkpoint. The value defined must be greater than 0,
```

Comment:
> Not sure this is correct English; maybe something like "The number of entities to process before triggering a checkpoint"

+1

> Also, why is it about the number of entities read? We don't want to consider an entity indexed and trigger a checkpoint if the entity has only been read, but not indexed, do we?

That was my mistake, nice point! I wrote "read" mistakenly because the trigger point is located at a reader method, `EntityReader#checkpointInfo()`. But we do want to trigger a checkpoint only after N entities being **indexed**.

From the §7.9 Batch Checkpoints: _"Checkpoints work naturally with chunk-oriented processing. The end of processing for each chunk is a natural point for taking a checkpoint."_, where "End of processing" means items having been read, processed, and written.